### ISSUE TITLE
ETCD2 Won't start on coreos latest

### ISSUE THREAD

**Issue by 0x4139** [coreos/etcd#5576]
```
Jun 07 12:46:24 x35b sudo[2483]:     core : TTY=pts/0 ; PWD=/home/core ; USER=root ; COMMAND=/bin/systemctl status etcd2.service
Jun 07 12:46:29 x35b sudo[2484]:     core : TTY=pts/0 ; PWD=/home/core ; USER=root ; COMMAND=/bin/systemctl restart etcd2.service
Jun 07 12:46:29 x35b polkitd[1589]: Registered Authentication Agent for unix-process:2484:88033 (system bus name :1.43 [/usr/bin/pkttyagent --notify-fd 5 --fallback], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8
Jun 07 12:46:29 x35b systemd[1]: Stopped etcd2.
-- Subject: Unit etcd2.service has finished shutting down
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit etcd2.service has finished shutting down.
Jun 07 12:46:29 x35b systemd[1]: Starting etcd2...
-- Subject: Unit etcd2.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit etcd2.service has begun starting up.
Jun 07 12:46:29 x35b etcd2[2489]: recognized and used environment variable ETCD_ADVERTISE_CLIENT_URLS=http://192.168.1.2:2379
Jun 07 12:46:29 x35b etcd2[2489]: recognized and used environment variable ETCD_DATA_DIR=/var/lib/etcd2
Jun 07 12:46:29 x35b etcd2[2489]: recognized and used environment variable ETCD_INITIAL_ADVERTISE_PEER_URLS=http://192.168.1.2:2380
Jun 07 12:46:29 x35b etcd2[2489]: recognized and used environment variable ETCD_LISTEN_CLIENT_URLS=http://192.168.1.2:2379
Jun 07 12:46:29 x35b etcd2[2489]: recognized and used environment variable ETCD_LISTEN_PEER_URLS=http://192.168.1.2:2380
Jun 07 12:46:29 x35b etcd2[2489]: recognized and used environment variable ETCD_NAME=f4320093d8b741fe8afbab7f40242066
Jun 07 12:46:29 x35b etcd2[2489]: etcd Version: 2.3.1
Jun 07 12:46:29 x35b etcd2[2489]: Git SHA: 2b67f52
Jun 07 12:46:29 x35b etcd2[2489]: Go Version: go1.5.3
Jun 07 12:46:29 x35b etcd2[2489]: Go OS/Arch: linux/amd64
Jun 07 12:46:29 x35b etcd2[2489]: setting maximum number of CPUs to 16, total number of available CPUs is 16
Jun 07 12:46:29 x35b etcd2[2489]: listening for peers on http://192.168.1.2:2380
Jun 07 12:46:29 x35b etcd2[2489]: listening for client requests on http://192.168.1.2:2379
Jun 07 12:46:29 x35b etcd2[2489]: stopping listening for client requests on http://192.168.1.2:2379
Jun 07 12:46:29 x35b etcd2[2489]: stopping listening for peers on http://192.168.1.2:2380
Jun 07 12:46:29 x35b etcd2[2489]: --initial-cluster must include f4320093d8b741fe8afbab7f40242066=http://192.168.1.2:2380 given --initial-advertise-peer-urls=http://192.168.1.2:2380
Jun 07 12:46:29 x35b etcd2[2489]: forgot to set --initial-cluster flag?
Jun 07 12:46:29 x35b etcd2[2489]: if you want to use discovery service, please set --discovery flag.
Jun 07 12:46:29 x35b systemd[1]: etcd2.service: Main process exited, code=exited, status=1/FAILURE
Jun 07 12:46:29 x35b systemd[1]: Failed to start etcd2.
-- Subject: Unit etcd2.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit etcd2.service has failed.
-- 
```
i have tried changing the discovery url and everything i have no clue why this isn't starting

**Comment by xiang90** [comment]
@0x4139 Can you please provide your etcd configuration? It seems that discovery-token is not provided from the log.

**Comment by khalaan** [comment]
I'm not sure if this is related or not but I also cannot start etcd2 on 1010.5.0.

journalctl -u etcd2

````
-- Reboot --
Jun 11 13:12:08 spengler.ghostbusters.sheltek.us systemd[1]: Starting etcd2...
Jun 11 13:12:08 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Control process exited, code=exited status=2
Jun 11 13:12:08 spengler.ghostbusters.sheltek.us systemd[1]: Failed to start etcd2.
Jun 11 13:12:08 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Unit entered failed state.
Jun 11 13:12:08 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Failed with result 'exit-code'.
Jun 11 13:12:18 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Service hold-off time over, scheduling restart.
Jun 11 13:12:18 spengler.ghostbusters.sheltek.us systemd[1]: Stopped etcd2.
Jun 11 13:12:18 spengler.ghostbusters.sheltek.us systemd[1]: Starting etcd2...
Jun 11 13:12:18 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Control process exited, code=exited status=2
Jun 11 13:12:18 spengler.ghostbusters.sheltek.us systemd[1]: Failed to start etcd2.
Jun 11 13:12:18 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Unit entered failed state.
Jun 11 13:12:18 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Failed with result 'exit-code'.
Jun 11 13:12:29 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Service hold-off time over, scheduling restart.
Jun 11 13:12:29 spengler.ghostbusters.sheltek.us systemd[1]: Stopped etcd2.
Jun 11 13:12:29 spengler.ghostbusters.sheltek.us systemd[1]: Starting etcd2...
Jun 11 13:12:29 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Control process exited, code=exited status=2
Jun 11 13:12:29 spengler.ghostbusters.sheltek.us systemd[1]: Failed to start etcd2.
Jun 11 13:12:29 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Unit entered failed state.
Jun 11 13:12:29 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Failed with result 'exit-code'.
Jun 11 13:12:39 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Service hold-off time over, scheduling restart.
Jun 11 13:12:39 spengler.ghostbusters.sheltek.us systemd[1]: Stopped etcd2.
Jun 11 13:12:39 spengler.ghostbusters.sheltek.us systemd[1]: Starting etcd2...
Jun 11 13:12:39 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Control process exited, code=exited status=2
Jun 11 13:12:39 spengler.ghostbusters.sheltek.us systemd[1]: Failed to start etcd2.
Jun 11 13:12:39 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Unit entered failed state.
Jun 11 13:12:39 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Failed with result 'exit-code'.
Jun 11 13:12:49 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Service hold-off time over, scheduling restart.
Jun 11 13:12:49 spengler.ghostbusters.sheltek.us systemd[1]: Stopped etcd2.
Jun 11 13:12:49 spengler.ghostbusters.sheltek.us systemd[1]: Starting etcd2...
Jun 11 13:12:49 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Control process exited, code=exited status=2
Jun 11 13:12:49 spengler.ghostbusters.sheltek.us systemd[1]: Failed to start etcd2.
Jun 11 13:12:49 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Unit entered failed state.
Jun 11 13:12:49 spengler.ghostbusters.sheltek.us systemd[1]: etcd2.service: Failed with result 'exit-code'.
```

and cloud-config looks like this:  
````
coreos:
  etcd2:
    name: spengler
    discovery: https://discovery.etcd.io/4c7ad265bd34ff638a021e401a85218f
    advertise-client-urls: http://192.168.1.232:2379,http://192.168.1.232:4001
    initial-advertise-peer-urls: http://192.168.1.232:2380
    listen-client-urls: http://0.0.0.0:2379,http://0.0.0.0:4001
    listen-peer-urls: http://192.168.1.232:2380
  units:
    - name: etcd2.service
      command: start
      drop-ins:
      - name: 50-network-config.conf
        content: |
          [Service]
          ExecStartPre=/usr/bin/etcdctl set /coreos.com/network/config '{"Network":"10.10.0.0/16", "Backend": {"Type": "vxlan"}}'
```  

Starting etcd2 by hand (or passing the parameters in by hand, rather) goes like this..


````
spengler ~ # !18
ETCD_DATA_DIR=/var/lib/etcd2 etcd2 --discovery https://discovery.etcd.io/0e7c08f5671e126e67b503df2adde3be --advertise-client-urls http://192.168.1.232:2379,http://192.168.1.232:4001 --initial-advertise-peer-urls http://192.168.1.232:2380 --listen-client-urls http://0.0.0.0:2379,http://0.0.0.0:4001 --listen-peer-urls http://192.168.1.232:2380 --name spengler
2016-06-12 02:30:34.040694 I | flags: recognized and used environment variable ETCD_DATA_DIR=/var/lib/etcd2
2016-06-12 02:30:34.040847 I | etcdmain: etcd Version: 2.3.1
2016-06-12 02:30:34.040857 I | etcdmain: Git SHA: 2b67f52
2016-06-12 02:30:34.040863 I | etcdmain: Go Version: go1.5.3
2016-06-12 02:30:34.040870 I | etcdmain: Go OS/Arch: linux/amd64
2016-06-12 02:30:34.040876 I | etcdmain: setting maximum number of CPUs to 16, total number of available CPUs is 16
2016-06-12 02:30:34.040935 N | etcdmain: the server is already initialized as member before, starting as etcd member...
2016-06-12 02:30:34.040988 I | etcdmain: listening for peers on http://192.168.1.232:2380
2016-06-12 02:30:34.041025 I | etcdmain: listening for client requests on http://0.0.0.0:2379
2016-06-12 02:30:34.041055 I | etcdmain: listening for client requests on http://0.0.0.0:4001
2016-06-12 02:30:34.057009 W | etcdserver: discovery token ignored since a cluster has already been initialized. Valid log found at "/var/lib/etcd2/member/wal"
2016-06-12 02:30:34.057039 I | etcdserver: name = spengler
2016-06-12 02:30:34.057048 I | etcdserver: data dir = /var/lib/etcd2
2016-06-12 02:30:34.057056 I | etcdserver: member dir = /var/lib/etcd2/member
2016-06-12 02:30:34.057064 I | etcdserver: heartbeat = 100ms
2016-06-12 02:30:34.057076 I | etcdserver: election = 1000ms
2016-06-12 02:30:34.057084 I | etcdserver: snapshot count = 10000
2016-06-12 02:30:34.057092 I | etcdserver: discovery URL= https://discovery.etcd.io/0e7c08f5671e126e67b503df2adde3be
2016-06-12 02:30:34.057116 I | etcdserver: advertise client URLs = http://192.168.1.232:2379,http://192.168.1.232:4001
2016-06-12 02:30:34.120314 I | etcdserver: restarting member c23abd567f2f8c7e in cluster e73568347adcdecd at commit index 6956
2016-06-12 02:30:34.120543 I | raft: c23abd567f2f8c7e became follower at term 2
2016-06-12 02:30:34.120564 I | raft: newRaft c23abd567f2f8c7e [peers: [], term: 2, commit: 6956, applied: 0, lastindex: 6956, lastterm: 2]
2016-06-12 02:30:34.123676 I | etcdserver: starting server... [version: 2.3.1, cluster version: to_be_decided]
2016-06-12 02:30:34.124044 E | etcdmain: failed to notify systemd for readiness: No socket
2016-06-12 02:30:34.124060 E | etcdmain: forgot to set Type=notify in systemd service file?
2016-06-12 02:30:34.136309 N | etcdserver: added member 950c7a0679d18d1 [http://192.168.1.231:2380] to cluster e73568347adcdecd
2016-06-12 02:30:34.136922 N | etcdserver: added member 2cf696424304f954 [http://192.168.1.233:2380] to cluster e73568347adcdecd
2016-06-12 02:30:34.150970 N | etcdserver: added member 97757695ec6e6be0 [http://192.168.1.230:2380] to cluster e73568347adcdecd
2016-06-12 02:30:34.151108 N | etcdserver: added local member c23abd567f2f8c7e [http://192.168.1.232:2380] to cluster e73568347adcdecd
2016-06-12 02:30:34.151556 N | etcdserver: set the initial cluster version to 2.3
2016-06-12 02:30:35.423983 I | raft: c23abd567f2f8c7e is starting a new election at term 2
2016-06-12 02:30:35.424079 I | raft: c23abd567f2f8c7e became candidate at term 3
2016-06-12 02:30:35.424094 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 3
2016-06-12 02:30:35.424131 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 3
2016-06-12 02:30:35.424145 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 3
2016-06-12 02:30:35.424160 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 3
2016-06-12 02:30:36.723939 I | raft: c23abd567f2f8c7e is starting a new election at term 3
2016-06-12 02:30:36.723986 I | raft: c23abd567f2f8c7e became candidate at term 4
2016-06-12 02:30:36.724003 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 4
2016-06-12 02:30:36.724020 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 4
2016-06-12 02:30:36.724036 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 4
2016-06-12 02:30:36.724071 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 4
2016-06-12 02:30:38.023935 I | raft: c23abd567f2f8c7e is starting a new election at term 4
2016-06-12 02:30:38.023982 I | raft: c23abd567f2f8c7e became candidate at term 5
2016-06-12 02:30:38.024007 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 5
2016-06-12 02:30:38.024037 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 5
2016-06-12 02:30:38.024057 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 5
2016-06-12 02:30:38.024078 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 5
2016-06-12 02:30:39.223873 I | raft: c23abd567f2f8c7e is starting a new election at term 5
2016-06-12 02:30:39.223918 I | raft: c23abd567f2f8c7e became candidate at term 6
2016-06-12 02:30:39.223933 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 6
2016-06-12 02:30:39.223952 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 6
2016-06-12 02:30:39.223971 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 6
2016-06-12 02:30:39.223989 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 6
2016-06-12 02:30:40.523901 I | raft: c23abd567f2f8c7e is starting a new election at term 6
2016-06-12 02:30:40.523969 I | raft: c23abd567f2f8c7e became candidate at term 7
2016-06-12 02:30:40.523982 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 7
2016-06-12 02:30:40.523996 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 7
2016-06-12 02:30:40.524008 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 7
2016-06-12 02:30:40.524020 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 7
2016-06-12 02:30:41.124121 E | etcdserver: publish error: etcdserver: request timed out
2016-06-12 02:30:41.353078 E | etcdhttp: got unexpected response error (etcdserver: request timed out)
2016-06-12 02:30:41.923952 I | raft: c23abd567f2f8c7e is starting a new election at term 7
2016-06-12 02:30:41.924012 I | raft: c23abd567f2f8c7e became candidate at term 8
2016-06-12 02:30:41.924030 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 8
2016-06-12 02:30:41.924051 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 8
2016-06-12 02:30:41.924064 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 8
2016-06-12 02:30:41.924074 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 8
2016-06-12 02:30:43.223947 I | raft: c23abd567f2f8c7e is starting a new election at term 8
2016-06-12 02:30:43.223994 I | raft: c23abd567f2f8c7e became candidate at term 9
2016-06-12 02:30:43.224010 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 9
2016-06-12 02:30:43.224025 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 9
2016-06-12 02:30:43.224037 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 9
2016-06-12 02:30:43.224049 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 9
2016-06-12 02:30:43.304928 E | etcdhttp: got unexpected response error (etcdserver: request timed out)
2016-06-12 02:30:44.524055 I | raft: c23abd567f2f8c7e is starting a new election at term 9
2016-06-12 02:30:44.524120 I | raft: c23abd567f2f8c7e became candidate at term 10
2016-06-12 02:30:44.524136 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 10
2016-06-12 02:30:44.524152 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 10
2016-06-12 02:30:44.524176 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 10
2016-06-12 02:30:44.524196 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 10
2016-06-12 02:30:45.040727 E | etcdhttp: got unexpected response error (etcdserver: request timed out) [merged 1 repeated lines in 1.74s]
2016-06-12 02:30:45.823899 I | raft: c23abd567f2f8c7e is starting a new election at term 10
2016-06-12 02:30:45.823959 I | raft: c23abd567f2f8c7e became candidate at term 11
2016-06-12 02:30:45.823977 I | raft: c23abd567f2f8c7e received vote from c23abd567f2f8c7e at term 11
2016-06-12 02:30:45.823996 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 950c7a0679d18d1 at term 11
2016-06-12 02:30:45.824014 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 2cf696424304f954 at term 11
2016-06-12 02:30:45.824033 I | raft: c23abd567f2f8c7e [logterm: 2, index: 6956] sent vote request to 97757695ec6e6be0 at term 11
^C2016-06-12 02:30:46.173021 N | osutil: received interrupt signal, shutting down...
```

Any suggestions where to start? Params match the cloud-config file...

Thanks.

**Comment by xiang90** [comment]
@0x4139 I am closing this out. We need more logging, reproduce steps to help you more. Feel free to reopen if this is still an issue for you and you have more information. Thanks!
